src/video/SDL_pixels.c

```diff
@@ -516,6 +572,12 @@ SDL_CreatePixelFormat(Uint32 pixel_format)
 {
     SDL_PixelFormat *format;
 
+    /* Only allow a known format, or explicitly SDL_PIXELFORMAT_UNKNOWN */
+    if (!SDL_IsPixelFormatValid(pixel_format)) {
```

Comment:
No, this is explicitly designed to support unknown formats that use the pixelformat layout. See SDL_GetMasksForPixelFormatEnum() for details.